File: util/android/src/main/java/com/dmi/util/base/BaseApplication.java
package com.dmi.util.base;

import android.app.Application;

import dagger.ObjectGraph;

public class BaseApplication extends Application {
    private ObjectGraph objectGraph;

    @SuppressWarnings("unchecked")
    @Override
    public void onCreate() {
        super.onCreate();
        objectGraph = createObjectGraph();
        if (objectGraph != null) {
            objectGraph.inject(this);
        }
    }

    protected ObjectGraph createObjectGraph() {
        return null;
    }

    protected ObjectGraph objectGraph() {
        return objectGraph;
    }
}

File: util/util-android/src/main/kotlin/com/dmi/util/base/BaseApplication.kt
package com.dmi.util.base

import android.app.Application

import dagger.ObjectGraph

open class BaseApplication : Application() {
    private lateinit var objectGraph: ObjectGraph

    @SuppressWarnings("unchecked")
    override fun onCreate() {
        super.onCreate()
//        objectGraph = createObjectGraph()
//        objectGraph.inject(this)
    }

    protected open fun createObjectGraph(): ObjectGraph {
        return ObjectGraph.create()
    }

    fun objectGraph(): ObjectGraph {
        return objectGraph
    }
}
